### ISSUE TITLE
Highlight comments that are too long

### ISSUE THREAD

**Issue by ukanuk** [KnoP-01/vim-tp#2]
It would be useful if line comments were highlighted when too long.

For example, when a file containing the line comment   ```  59:  !*****  Check if the Gripper is empty ;``` is imported into the robot, it is truncated after the `e` of `empty`. If this were highlighted in vim, I could change it there and not worry about losing comments after import into the robot.

![image](https://user-images.githubusercontent.com/59303202/108377761-fa802780-71c9-11eb-9ff1-880ec0a6d571.png)

This seems possible based on a cursory Google search revealing:
* https://stackoverflow.com/questions/16655031/highlight-lines-that-are-too-long
* https://github.com/aklt/vim-line_length

**Comment by ukanuk** [comment]
Note there are two spaces in front of ```59: !***** Check if the Gripper is empty ;```, but Github is stripping them when I submit the comment.

**Comment by ukanuk** [comment]
And for a temporary workaround, one can use `:set colorcolumn=41`, although it highlights all lines rather than just comments.

**Comment by KnoP-01** [comment]
@ukanuk: Many thanks for your Input!

Note that this repo is a fork of [onerobotics/vim-tp](https://github.com/onerobotics/vim-tp). 
I was just not happy, so here is my fork. I have not kept track of any changes over there since.
My main focus is on my Rapid and KRL plugins. I have no time to treat this one with the same energy.

`call matchadd('colorcolumn','\%41v')` highlights only lines that are longer than 41 columns with a colorcolumn. Vim >8 or 8.1 may be required for matchadd.

However, I will put something in the [dev tree](https://github.com/KnoP-01/vim-tp/tree/dev) of this repo to address your suggestion. See if it works for you. I may merge this into master after a while.

The same could be done for the `NN:  MESSAGE[   ] ;` command. However, this kind of error highlighting tend to get expensive on the performance side. Maybe I will wrap this in an option some day.

**Comment by KnoP-01** [comment]
@ukanuk: Feel free to open a new issues if you have any trouble with this feature or any more suggestions.

**Comment by ukanuk** [comment]
@KnoP-01 That update is working well, thank you! I did originally install the onerobotics repo, but this fork seemed more fully-featured and actively developed (even if I'm not necessarily using the more advanced features yet).

**Comment by KnoP-01** [comment]
@ukanuk: Not much advanced features here, apart from setting 'suffixesadd'=.ls which makes gf and \<CTRL\>+Wf possible on the routine name of a CALL and the mapping \<leader\>dc (delete comments) to delete comments (like P[1:home]->P[1], DI[1:Foo]->DI[1] and so on) apart from label comments. I find it sometimes more readable without comments. The latter is not extensively tested. Use at own risk ;) It will fail if the comment contains [ or ].

Maybe you can answer the question: Is .pe a valid file name extension for Fanuc TP files? In that case I should add .pe as well to 'suffixesadd'. What about capitol letter extensions like .PE and .LS?

Any other feature requests? Maybe, automatically add a line number on i_\<CR\>, o and O?

Maybe with your help, I manage to get this thing more polished.

**Comment by ukanuk** [comment]
Adding *.LS would be great, that's what ROBOGUIDE exports (not *.ls).

I'm pretty new to FANUC, but I will let you know if I think of any other feature requests! I am not familiar with the the .pe/.PE extension.

I don't think auto-adding numbers to new lines would be particularly useful. My workflow is exporting LS files from ROBOGUIDE, making text edits to the file, then re-importing to ROBOGUIDE. That re-import into ROBOGUIDE already adds/fixes line numbers, so every time I go from the robot back to the text editor the line numbers are automatically fixed. Once this project is complete, the "master copy" of programs will live on the robot with perfect line numbers, and I'll delete the files I worked on with my text editor. And if you weren't going between the robot and text files which forced you to keep having line numbers, then you wouldn't want to add line numbers to the LS file anyway.

> https://www.onerobotics.com/posts/2014/writing-tp-programs-by-hand---pros-and-cons/
>  You can optionally include a line number before the colon, but I think including line numbers in your source code is a tragedy.

**Comment by KnoP-01** [comment]
.PE is the extension that you get if you translate .TP to ASCII with tpptrane.exe.  I added .pe to both, ftdetect and 'suffixesadd'.

You mean you can omit the line number and the colon completely? I have neither a RG license nor a robot with ASCII import atm to test. I know that the numbers don't have to be in order. The idea was to add a dummy line number like 999: or so.

**Comment by ukanuk** [comment]
You do need the colon, but do not need the line number. I just tested it to double-check, and it works both with and without spaces in front of the colon. 

Although in testing this, I notice that the comment syntax highlighting breaks when there's no number at the start of the line.

Correct:
![image](https://user-images.githubusercontent.com/59303202/108575704-b5024e00-72e0-11eb-8275-b0f9c56e296e.png)

Broken:
![image](https://user-images.githubusercontent.com/59303202/108575718-c51a2d80-72e0-11eb-8287-4fb74535a1bc.png)

**Comment by KnoP-01** [comment]
I'll fix that, probably on Monday.

**Comment by KnoP-01** [comment]
not yet merged to master

**Comment by ukanuk** [comment]
I just briefly tested the update, and it seems to be working perfectly!

**Comment by KnoP-01** [comment]
Thanks for your feedback. Merged into Master branch.
